sound/soc/sof/core.c

```diff
@@ -356,12 +369,15 @@ int snd_sof_device_remove(struct device *dev)
 			dev_warn(dev, "error: %d failed to prepare DSP for device removal",
 				 ret);
 
+		snd_sof_unregister_clients(sdev);
 		snd_sof_fw_unload(sdev);
 		snd_sof_ipc_free(sdev);
 		snd_sof_free_debug(sdev);
 		snd_sof_free_trace(sdev);
 	}
 
+	ida_destroy(&sof_ipc_test_client_ida);
```

Comment:
why is this here in the SOF core? This should be in the client, not here.

also isn't this imbalanced?

Reply:
@plbossart this cant be done in the client. For example both the ipc_test clients using this IDA and when they are unregistered, they free the id allocated to them. So the core has to do the ida_destroy() after all the clients are unregistered.

Reply:
fixed